### ISSUE TITLE
volume must be a mapping, not an array.

### ISSUE THREAD

**Issue by ehelms** [ansible/ansible-container#544]
##### ISSUE TYPE
 - Bug Report

##### container.yml 
```
---
version: "2"
settings:
  conductor_base: centos:7
defaults:
  POSTGRES_USER: foreman
  POSTGRES_PASSWORD: foreman
  POSTGRES_DB: foreman
  POSTGRES_PORT: 5432
  MONGODB_USER: admin
  MONGODB_PASSWORD: admin
  MONGODB_DATABASE: pulp_database
  MONGODB_PORT: 27017
services:
  foreman-base:
    from: centos:7
    roles:
      - epel-repositories
      - role: puppet-repositories
        puppet_repositories_version: 4
      - foreman-repositories
      - katello-repositories
      - foreman
  foreman:
    from: openshift-foreman-base:latest
    command: ['/usr/bin/start-foreman.sh']
    entrypoint: ['/usr/bin/entrypoint.sh']
    environment:
      - POSTGRES_DB={{ POSTGRES_DB }}
      - POSTGRES_USER={{ POSTGRES_USER }}
      - POSTGRES_PASSWORD={{ POSTGRES_PASSWORD }}
      - SEED_ADMIN_USER=admin
      - SEED_ADMIN_PASSWORD=changeme
      - SEED_ORGANIZATION=Default Organization
      - SEED_LOCATION=Raleigh
    ports:
      - 8080:8080
    expose:
      - 8080
  foreman-tasks:
    from: openshift-foreman-base:latest
    roles:
      - foreman-tasks
    command: ['/usr/bin/start-foreman-tasks.sh']
    entrypoint: ['/usr/bin/entrypoint.sh']
    environment:
      - POSTGRES_DB={{ POSTGRES_DB }}
      - POSTGRES_USER={{ POSTGRES_USER }}
      - POSTGRES_PASSWORD={{ POSTGRES_PASSWORD }}
  foreman-proxy:
    from: centos:7
    roles:
      - epel-repositories
      - foreman-repositories
      - foreman-proxy
    command: ['/usr/bin/start-foreman-proxy.sh']
    expose:
      - 8080
  foreman-proxy-register:
    from: centos:7
    roles:
      - epel-repositories
      - foreman-repositories
      - foreman-proxy
    command: ['/usr/bin/register.sh']
    environment:
      - FOREMAN_PROXY_SERVICE_HOST=foreman_proxy
      - FOREMAN_PROXY_SERVICE_PORT=8080
      - FOREMAN_SERVICE_HOST=foreman
      - FOREMAN_SERVICE_PORT=8080
    depends_on:
      - foreman
      - foreman-proxy
  cp-app:
    from: centos:7
    roles:
      - epel-repositories
      - katello-repositories
      - candlepin
    environment:
      - POSTGRES_DB={{ POSTGRES_DB }}
      - POSTGRES_USER={{ POSTGRES_USER }}
      - POSTGRES_PASSWORD={{ POSTGRES_PASSWORD }}
      - POSTGRES_PORT={{ POSTGRES_PORT }}
      - POSTGRES_SERVICE=postgres
      - QPID_SERVICE=qpid
      - QPID_PORT=5672
    expose:
      - 8080
    command: ['/usr/bin/start_candlepin.sh']
    entrypoint: ['/usr/bin/entrypoint.sh']
    options:
      kube:
        runAsUser: 1000
  qpid:
    from: centos:7
    roles:
      - epel-repositories
      - qpid
    command: ['/usr/bin/startup.sh']
    expose:
      - 5672
  pulp:
    from: centos:7
    roles:
      - epel-repositories
      - katello-repositories
      - pulp
    command: ['/usr/bin/start_httpd.sh']
    entrypoint: ['/usr/bin/entrypoint.sh']
    expose:
      - 8080
    depends_on:
      - qpid
      - mongodb
  pulp-worker:
    from: centos:7
    roles:
      - epel-repositories
      - katello-repositories
      - pulp
      - pulp-worker
    entrypoint: ['/usr/bin/entrypoint.sh']
    depends_on:
      - qpid
      - pulp
    depends_on:
      - mongodb
    volumes:
      - pulp-data:/var/lib/pulp
      - puppet-data:/etc/puppet
    options:
      openshift:
        persistent_volume_claims:
          - volume_name: pulp-data
            claim_name: pulp-data
            access_modes:
              - ReadWriteMany
          - volume_name: puppet-data
            claim_name: puppet-data
            access_modes:
              - ReadWriteMany
  pulp-resource-manager:
    from: centos:7
    roles:
      - epel-repositories
      - katello-repositories
      - pulp
      - pulp-resource-manager
    depends_on:
      - qpid
      - pulp
    entrypoint: ['/usr/bin/entrypoint.sh']
  pulp-celerybeat:
    from: centos:7
    roles:
      - epel-repositories
      - katello-repositories
      - pulp
      - pulp-celerybeat
    entrypoint: ['/usr/bin/entrypoint.sh']
    depends_on:
      - qpid
      - pulp
  puppet:
    from: puppet/puppetserver
    volumes:
      - puppet-data:/etc/puppet
    expose:
      - 8140
    options:
      openshift:
        persistent_volume_claims:
          - volume_name: puppet-data
            claim_name: puppet-data
            access_modes:
              - ReadWriteMany
  postgres:
    from: ansible/postgresql:latest
    environment:
      - POSTGRES_DB={{ POSTGRES_DB }}
      - POSTGRES_USER={{ POSTGRES_USER }}
      - POSTGRES_PASS={{ POSTGRES_PASSWORD }}
      - PGDATA=/var/lib/pgsql/data/userdata
    volumes:
      - postgres-data:/var/lib/pgsql/data
    expose:
      - "{{ POSTGRES_PORT }}"
    options:
      openshift:
        persistent_volume_claims:
          - volume_name: postgres-data
            claim_name: postgres-data
            access_modes:
              - ReadWriteMany
  mongodb:
    from: centos/mongodb-26-centos7:latest
    environment:
      - MONGODB_USER={{ MONGODB_USER }}
      - MONGODB_PASSWORD={{ MONGODB_PASSWORD }}
      - MONGODB_DATABASE={{ MONGODB_DATABASE }}
      - MONGODB_ADMIN_PASSWORD={{ MONGODB_PASSWORD }}
    volumes:
      - mongodb-data:/var/lib/mongodb/data
    expose:
      - "{{ MONGODB_PORT }}"
    options:
      openshift:
        persistent_volume_claims:
          - volume_name: mongodb-data
            claim_name: mongodb-data
            access_modes:
              - ReadWriteMany
volumes:
  postgres-data: {}
  mongodb-data: {}
  puppet-data: {}
  pulp-data: {}
registries:
  oc-cluster:
    url: http://172.30.1.1:5000
    namespace: foreman
  # Add optional registries used for deployment. For example:
  #  google:
  #    url: https://gcr.io
  #    namespace: my-cool-project-xxxxxx
```

##### OS / ENVIRONMENT
```
Ansible Container, version 0.9.1rc0
Linux, war.eagle, 4.6.6-200.fc23.x86_64, #1 SMP Wed Aug 10 23:13:35 UTC 2016, x86_64
2.7.11 (default, Jul  8 2016, 19:45:00) 
[GCC 5.3.1 20160406 (Red Hat 5.3.1-6)] /usr/bin/python
{
  "ContainersPaused": 0, 
  "Labels": null, 
  "DefaultRuntime": "runc", 
  "CgroupDriver": "cgroupfs", 
  "ContainersRunning": 6, 
  "NGoroutines": 56, 
  "Swarm": {
    "Managers": 0, 
    "ControlAvailable": false, 
    "NodeID": "", 
    "Cluster": {
      "Spec": {
        "Raft": {}, 
        "CAConfig": {}, 
        "Dispatcher": {}, 
        "Orchestration": {}, 
        "TaskDefaults": {}
      }, 
      "Version": {}, 
      "ID": "", 
      "CreatedAt": "0001-01-01T00:00:00Z", 
      "UpdatedAt": "0001-01-01T00:00:00Z"
    }, 
    "Nodes": 0, 
    "Error": "", 
    "RemoteManagers": null, 
    "LocalNodeState": "inactive", 
    "NodeAddr": ""
  }, 
  "LoggingDriver": "json-file", 
  "OSType": "linux", 
  "HttpProxy": "", 
  "Runtimes": {
    "runc": {
      "path": "docker-runc"
    }
  }, 
  "DriverStatus": [
    [
      "Pool Name", 
      "docker-253:3-17170446-pool"
    ], 
    [
      "Pool Blocksize", 
      "65.54 kB"
    ], 
    [
      "Base Device Size", 
      "10.74 GB"
    ], 
    [
      "Backing Filesystem", 
      "xfs"
    ], 
    [
      "Data file", 
      "/dev/loop0"
    ], 
    [
      "Metadata file", 
      "/dev/loop1"
    ], 
    [
      "Data Space Used", 
      "26.71 GB"
    ], 
    [
      "Data Space Total", 
      "107.4 GB"
    ], 
    [
      "Data Space Available", 
      "80.66 GB"
    ], 
    [
      "Metadata Space Used", 
      "28.62 MB"
    ], 
    [
      "Metadata Space Total", 
      "2.147 GB"
    ], 
    [
      "Metadata Space Available", 
      "2.119 GB"
    ], 
    [
      "Thin Pool Minimum Free Space", 
      "10.74 GB"
    ], 
    [
      "Udev Sync Supported", 
      "true"
    ], 
    [
      "Deferred Removal Enabled", 
      "false"
    ], 
    [
      "Deferred Deletion Enabled", 
      "false"
    ], 
    [
      "Deferred Deleted Device Count", 
      "0"
    ], 
    [
      "Data loop file", 
      "/home/docker/devicemapper/devicemapper/data"
    ], 
    [
      "Metadata loop file", 
      "/home/docker/devicemapper/devicemapper/metadata"
    ], 
    [
      "Library Version", 
      "1.02.109 (2015-09-22)"
    ]
  ], 
  "OperatingSystem": "Fedora 23 (Workstation Edition)", 
  "Containers": 8, 
  "HttpsProxy": "", 
  "BridgeNfIp6tables": true, 
  "MemTotal": 16700760064, 
  "Driver": "devicemapper", 
  "IndexServerAddress": "https://index.docker.io/v1/", 
  "ClusterStore": "", 
  "ExecutionDriver": "", 
  "SystemStatus": null, 
  "OomKillDisable": true, 
  "ClusterAdvertise": "", 
  "SystemTime": "2017-05-24T08:56:00.348733351-04:00", 
  "Name": "war.eagle", 
  "CPUSet": true, 
  "RegistryConfig": {
    "InsecureRegistryCIDRs": [
      "172.30.0.0/16", 
      "127.0.0.0/8"
    ], 
    "IndexConfigs": {
      "docker.io": {
        "Official": true, 
        "Name": "docker.io", 
        "Secure": true, 
        "Mirrors": null
      }
    }, 
    "Mirrors": null
  }, 
  "SecurityOptions": [
    "seccomp"
  ], 
  "ContainersStopped": 2, 
  "NCPU": 8, 
  "NFd": 55, 
  "Architecture": "x86_64", 
  "KernelMemory": true, 
  "CpuCfsQuota": true, 
  "Debug": false, 
  "ID": "PP7J:YWUF:VJG7:OPQQ:AVCZ:BEIA:XEX2:BB7W:ELOD:I2QM:O472:Y2MB", 
  "IPv4Forwarding": true, 
  "KernelVersion": "4.6.6-200.fc23.x86_64", 
  "BridgeNfIptables": true, 
  "NoProxy": "", 
  "LiveRestoreEnabled": false, 
  "ServerVersion": "1.12.6", 
  "CpuCfsPeriod": true, 
  "ExperimentalBuild": false, 
  "MemoryLimit": true, 
  "SwapLimit": true, 
  "Plugins": {
    "Volume": [
      "local"
    ], 
    "Network": [
      "host", 
      "bridge", 
      "null", 
      "overlay"
    ], 
    "Authorization": null
  }, 
  "Images": 163, 
  "DockerRootDir": "/home/docker", 
  "NEventsListener": 0, 
  "CPUShares": true
}
{
  "KernelVersion": "4.6.6-200.fc23.x86_64", 
  "Os": "linux", 
  "BuildTime": "2017-01-10T20:21:11.017972579+00:00", 
  "ApiVersion": "1.24", 
  "Version": "1.12.6", 
  "GitCommit": "78d1802", 
  "Arch": "amd64", 
  "GoVersion": "go1.6.4"
}
```

##### SUMMARY
Worth noting is that this is a conversion from 0.3 to 0.9. I run into the following error output when running `ansible-container run`:

```
Engine integration loaded. Preparing run.	engine=Docker™ daemon
Verifying service image	service=foreman-base
Verifying service image	service=foreman-tasks
Verifying service image	service=foreman-proxy
Verifying service image	service=foreman-proxy-register
Verifying service image	service=cp-app
Verifying service image	service=qpid
Verifying service image	service=pulp
Verifying service image	service=pulp-worker
Verifying service image	service=pulp-resource-manager
Verifying service image	service=pulp-celerybeat
WARNING	Image puppet/puppetserver for service puppet not found. An attempt will be made to pull it.
WARNING	Image ansible/postgresql:latest for service postgres not found. An attempt will be made to pull it.
WARNING	Image centos/mongodb-26-centos7:latest for service mongodb not found. An attempt will be made to pull it.

PLAY [localhost] ***************************************************************

TASK [docker_service] **********************************************************
fatal: [localhost]: FAILED! => {"changed": false, "failed": true, "msg": "Configuration error - In file '/tmp/ansibleOujUMm/docker-compose.yml', volume must be a mapping, not an array."}

msg: Configuration error - In file '/tmp/ansibleOujUMm/docker-compose.yml', volume must be a mapping, not an array.

PLAY RECAP *********************************************************************
localhost                  : ok=0    changed=0    unreachable=0    failed=1

All services running.	playbook_rc=2
Conductor terminated. Cleaning up.	command_rc=0 conductor_id=e6a3fc68d2d8704596e3b6d08b948b4532f80581bf17160c795a8f487262b616 save_container=False
```

**Comment by chouseknecht** [comment]
This is fixed in PR #532 by commit [a18b85fb29667269be56a795ed212571cf5cc1cd](https://github.com/ansible/ansible-container/pull/532/commits/a18b85fb29667269be56a795ed212571cf5cc1cd)

**Comment by ehelms** [comment]
@chouseknecht Thanks for the pointer. I switched to using your fork and branch and installed from source. I am still hitting this error though.

**Comment by ehelms** [comment]
Looking at it more, I think it was perhaps holding on to an older version of the conductor container. After some cleanup I am not seeing this anymore using your PR.

**Comment by j00bar** [comment]
@ehelms Don't sweat it. I did the exact same thing during a live demo at a conference this morning. 🙄

**Comment by j00bar** [comment]
@ehelms It's fixed in 0.9.1/devel now too, just differently than @chouseknecht's branch.
